src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs

```diff
@@ -6,9 +6,27 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
+        internal static bool UseIcu { get; private set; } = GetUseIcuMode();
+
         private static bool GetGlobalizationInvariantMode()
         {
             return GetInvariantSwitchValue();
         }
+
+        private static bool GetUseIcuMode()
+        {
+            bool useNlsValue = GetUseNlsSwitchValue();
+            if (!Invariant)
```

Comment:
This is fragile pattern - it only worth if the Invariant is initialized before IcuMode.

Reply:
Wouldn't this force Invariant to be initialized? So you suggest instead calling `GetInvariantSwitchValue` here?

Reply:
All static fields are initialized in the same static constructor. Accessing the static fields from the static constructors works only if the static field was initialized already.

The happens to work fine because GlobalizationMode.cs is before GlobalizationMode.Windows.cs in the .csproj file; but depending on this order is a fragile pattern that often leads to bugs.

Reply:
I do not think we would want to call `GetInvariantSwitchValue` twice. I would put both the properties into the same file, next to each other, with a comment that the order is important.

Reply:
Ok, I'll refactor this as suggested. Thanks.